eesp.org

```diff
@@ -693,6 +693,45 @@ a KDF MUST be used used to derive a separate key for each anti-replay sequence
 number subspace. In this case, the full 64 bits of each anti-replay
 sequence number subspace can be used.
 
+Sub SAs can be created "on the fly" within the kernel IPsec subsystem.
+Sub SAs streamline traffic flow management, reduce overhead, and
+enable more efficient lifecycle operations.
+
+A pair of EESP SAs combined with multiple unidirectional Sub SAs
+provides a more flexible approach to carrying  asymmetric traffic
+patterns, particularly in high-speed environments.
+Sub SAs reduces overhead, improves resource utilization, and enhances
+scalability for large-scale deployments. In many use cases, several
+uni directinal SAs used, while others are unused which can result
+in unnecessary overhead for SA management, rekeying, and resource
+consumption. Furthermore, using multiple bidirectional Child SAs for
+granular traffic flows often leads to additional setup delays and
+complex lifetime management. This inefficiency is particularly acute
+in high-throughput or low-latency environments, where rapid setup and
+teardown of SAs is essential to maintain performance.
+
+Each Sub SA is identified by a Sub SA ID, which MUST be carried in
+each EESP packet in the Session ID field—consistent with the
+negotiation of the EESP Child SA. This Sub SA ID is used to derive a
+ unique key.
+
+Implementations using hierarchical key derivation, particularly with
+hardware offload, MAY derive Sub SA keys dynamically on a per-packet
+basis.This mitigates the risk of data-plane performance degradation
```

Comment:
Won't the key derivation be defined in this document? What is the "implementations using hierarchical key derivation" referring to? And why would the dynamic key derivation be limited to those specifically?